midterm/Neumannova_midterm.R

```diff
@@ -0,0 +1,360 @@
+# vector with TRUE if indicator is contained in the dataframe
+# call this dataframe available_series
+
+indicators = c('ph_seq','ffpos', 'pppos','region','gereg', 'lfstat','a_lfsr','h_year','a_age',
+              'race','prdtrace','hispanic','prdthsp','educ_cat','a_hge','lths','hs','rsnnottw' ,
+              'so_col','college', 'grad' ,'semp_yn','a_sex','fwsval',
+              'not_working','selfempl','wages' ,'hrsly','hrslt','female' ,'married',
+              'divorced', 'children')
+
+function_available_series <- function(indicator, data) {
+  available <- indicator %in% colnames(data)
+  columns <- cbind(indicator, available)
+  as.data.frame(columns)
+}
+
+avai_series <- function_available_series(indicators, df)
+available_series <- as.data.frame(avai_series)
+available_series
+
+# I prepare a vector of desired columns names to work with them in the following problems
+
+des_columns <- available_series %>%
+  filter(available == TRUE)
+desired_columns <- as.vector(des_columns[[1]])
+desired_columns
+
+#1.3)
+# Create a slice of the dataframe for those indicators for which column available is TRUE. 
+# Call the result df_slice
+# You can either do that by creating a function or without creating a function
+
+function_slice_available <- function(data, columns) {
+  data[, which(colnames(data) %in% columns)]
+}
+
+df_slice_notordered <- function_slice_available(df, desired_columns)
+
+#the following line changes the order of columns in df_slice_notordered so that I can easily rename them in the task 3
+df_slice <- df_slice_notordered[desired_columns]
+
+
+#############################################################################################
+# Task 2
+#############################################################################################
+# You're gonna continue working with the slice only. Feel free to rename it and to remove any unnecessary
+# variables
+
+
+#2.1) Name of the columns are not really informative.Rename columns using col_names vector
+col_names = c("household_id","family_id","individual_id",'region','employment_status',
+              'year','age','race','hispanic','self_employed','sex','family_income')
+
+names(df_slice) <- col_names
+summary(df_slice)
+
+
+# 2.2)
+# Create a function that has dataframe df as an input and returns you the number of households,
+# number of families and number of individuals within the dataframe df
+
+#using magrittr
+
+mag_households <- df_slice %>%
+  count(household_id) %>%
+  nrow()
+
+mag_families <- df_slice %>%
+  group_by(household_id) %>%
+  count(family_id) %>%
+  nrow()
+  
+mag_individuals <- df_slice %>%
+  group_by(household_id, family_id) %>%
+  count(individual_id) %>%
+  nrow()
+
+mag_households
+mag_families
+mag_individuals
+  
+#not using pipelines - generally
+unique_households <- count(df_slice, household_id)
+households <- nrow(unique_households)
+households_grouping <- group_by(df_slice, household_id)
```

Comment:
You can do that in a single line by
unique_households <- nrow(count(df_slice, household_id))